test/e2e/util.go

```diff
@@ -129,3 +132,17 @@ func validateInferenceWithClient(t *testing.T, client *openai.Client, modelName
 	assert.NotEmpty(t, chatCompletion.Choices, "chat completion has no choices returned")
 	assert.NotNil(t, chatCompletion.Choices[0].Message.Content, "chat completion has no message returned")
 }
+
+func validateAllPodsAreReady(t *testing.T, client *kubernetes.Clientset, expectedPodCount int) {
+	allPodsReady := false
+	for i := 0; i <= 3; i++ {
+		podlist, err := client.CoreV1().Pods("default").List(context.Background(), v1.ListOptions{})
+		if err == nil && expectedPodCount == len(utils.FilterActivePods(podlist.Items)) {
+			allPodsReady = true
+			t.Log("all pods are ready", len(utils.FilterActivePods(podlist.Items)))
+			break
+		}
+		time.Sleep(1 * time.Second)
```

Comment:
I believe https://github.com/onsi/ginkgo is more solid here with like `Eventually` semantic.